graylog2-server/src/main/java/org/graylog/events/processor/aggregation/AggregationEventProcessor.java

```diff
@@ -229,6 +251,7 @@ private boolean satisfiesConditions(AggregationKeyResult keyResult) {
             // TODO: Do we have to set any other event fields here?
             event.setTimerangeStart(parameters.timerange().getFrom());
             event.setTimerangeEnd(parameters.timerange().getTo());
+            sourceStreams.forEach(event::addSourceStream);
```

Comment:
wouldn't sth like `event.setSourceStreams(sourceStreams)` be more efficient?

Reply:
Yes, I think it would. The idea of not providing `set*` style methods for streams and source-streams was to avoid overriding all streams by accident.